include/modem/lte_lc.h

```diff
@@ -204,9 +207,115 @@ enum lte_lc_pdp_type {
 };
 
 enum lte_lc_pdn_auth_type {
-	LTE_LC_PDN_AUTH_TYPE_NONE = 0,
-	LTE_LC_PDN_AUTH_TYPE_PAP = 1,
-	LTE_LC_PDN_AUTH_TYPE_CHAP = 2,
+	LTE_LC_PDN_AUTH_TYPE_NONE	= 0,
+	LTE_LC_PDN_AUTH_TYPE_PAP	= 1,
+	LTE_LC_PDN_AUTH_TYPE_CHAP	= 2,
+};
+
+enum lte_lc_energy_estimate {
+	LTE_LC_ENERGY_CONSUMPTION_EXCESSIVE	= 5,
+	LTE_LC_ENERGY_CONSUMPTION_INCREASED	= 6,
+	LTE_LC_ENERGY_CONSUMPTION_NORMAL	= 7,
+	LTE_LC_ENERGY_CONSUMPTION_REDUCED	= 8,
+	LTE_LC_ENERGY_CONSUMPTION_EFFICIENT	= 9,
+};
+
+enum lte_lc_tau_triggered {
+	LTE_LC_CELL_IN_TAI_LIST		= 0,
+
+	/* Cell not in TAI list, TAU will be triggered. */
+	LTE_LC_CELL_NOT_IN_TAI_LIST	= 1,
+	LTE_LC_CELL_UNKNOWN		= UINT8_MAX
+};
+
+enum lte_lc_ce_level {
+	LTE_LC_CE_LEVEL_NO_REPETITION		= 0,
+	LTE_LC_CE_LEVEL_LOW_REPETITION		= 1,
+	LTE_LC_CE_LEVEL_MEDIUM_REPETITION	= 2,
+	LTE_LC_CE_LEVEL_HIGH_REPETITION		= 3,
+	LTE_LC_CE_LEVEL_UNKNOWN			= UINT8_MAX,
+};
+
+/** @brief Connection evalution parameters.
+ *
+ *  @note For more information on the various connection evaluation parameters, refer to the
+ *	  "nRF91 AT Commands - Command Reference Guide".
+ */
+struct lte_lc_conn_eval_params {
+	/* RRC connection state during measurements. */
+	enum lte_lc_rrc_mode rrc_state;
+
+	/* Relative estimated energy consumption for data transmission compared to nominal
+	 * consumption.
+	 */
+	enum lte_lc_energy_estimate energy_estimate;
+
+	/* Is the evaluated cell part of the current TAI list. */
+	enum lte_lc_tau_triggered tau_trig;
+
+	/* Coverage Enhancement level for PRACH depending on RRC state measured in.
+	 *
+	 * RRC IDLE: CE level is estimated based on RSRP and network configueration.
+	 *
+	 * RRC CONNECTED: Currently configured CE level.
+	 *
+	 * CE levels 0-2 are supported in NB-IoT.
+	 * CE levels 0-1 are supported in LTE-M.
+	 */
+	enum lte_lc_ce_level ce_level;
+
+	/* EARFCN for given cell where EARFCN is per 3GPP TS 36.101. */
+	int earfcn;
+
+	/* Reduction in power density in dB. */
+	int16_t dl_pathloss;
+
+	/* Current RSRP level at time of report. */
+	int16_t rsrp;
```

Comment:
LTE LC does not currently do any conversion on such values. Given the nature of how these values are specified in the AT command doc I thought it may be best to leave it up to the application on they should interpreted/handled. That is also why they do not have units documented.

Reply:
If you find it sufficient I can document the ranges for the concerning parameters (rsrp, rsrq and snr) reflecting the AT command spec.

Reply:
I'm not sure if we have had similar functionality in LTE LC earlier and how it has been implemented, but just though that it would be easier for the application developers if the values would already be converted to dB or dBm values. I guess we can thank 3GPP for the values being like this in the AT commands...

If we don't want to convert the values in LTE LC, then we should either describe the mapping or refer to the AT command spec here, I guess.